src/frontend/components/Felleskomponenter/UtenlandsoppholdModal/utenlandsoppholdSpråkUtils.ts

```diff
@@ -0,0 +1,87 @@
+import { IBarnMedISøknad } from '../../../typer/søknad';
+import { EUtenlandsoppholdÅrsak } from '../../../typer/utenlandsopphold';
+import {
+    landFeilmeldingSpråkIdsSøker,
+    landLabelSpråkIdsBarn,
+    landLabelSpråkIdsSøker,
+    årsakSpråkIdsBarn,
+    årsakSpråkIdsSøker,
+} from './spørsmål';
+
+export const årsakFeilmeldingSpråkId = (barn?: IBarnMedISøknad): string =>
+    barn ? 'ombarnet.beskriveopphold.feilmelding' : 'modal.beskriveopphold.feilmelding';
+
+export const årsakLabelSpråkId = (barn?: IBarnMedISøknad): string =>
+    barn ? 'ombarnet.beskriveopphold.spm' : 'modal.beskriveopphold.spm';
+
+export const årsakSpråkId = (årsak: EUtenlandsoppholdÅrsak | '', barn?: IBarnMedISøknad): string =>
+    barn ? årsakSpråkIdsBarn[årsak] : årsakSpråkIdsSøker[årsak];
+
+export const landLabelSpråkId = (årsak: EUtenlandsoppholdÅrsak | '', barn?: IBarnMedISøknad) =>
+    barn ? landLabelSpråkIdsBarn[årsak] : landLabelSpråkIdsSøker[årsak];
+
+export const landFeilmeldingSpråkId = (
+    årsak: EUtenlandsoppholdÅrsak | '',
+    barn?: IBarnMedISøknad
+) => (barn ? landFeilmeldingSpråkIdsBarn[årsak] : landFeilmeldingSpråkIdsSøker[årsak]);
+
+export const landFeilmeldingSpråkIdsBarn: Record<EUtenlandsoppholdÅrsak, string> = {
+    [EUtenlandsoppholdÅrsak.FLYTTET_PERMANENT_TIL_NORGE]:
+        'ombarnet.hvilketlandflyttetfra.feilmelding',
+    [EUtenlandsoppholdÅrsak.FLYTTET_PERMANENT_FRA_NORGE]:
+        'ombarnet.hvilketlandflyttettil.feilmelding',
+    [EUtenlandsoppholdÅrsak.HAR_OPPHOLDT_SEG_UTENFOR_NORGE]:
+        'ombarnet.hvilketlandoppholdti.feilmelding',
+    [EUtenlandsoppholdÅrsak.OPPHOLDER_SEG_UTENFOR_NORGE]:
+        'ombarnet.hvilketlandoppholderi.feilmelding',
+};
+
+export const fraDatoFeilmeldingSpråkId = (
+    årsak: EUtenlandsoppholdÅrsak | '',
+    barn?: IBarnMedISøknad
+) => {
+    switch (årsak) {
+        case EUtenlandsoppholdÅrsak.FLYTTET_PERMANENT_FRA_NORGE:
```

Comment:
Finnes det noen standard på om man skal bruke switch/case eller ikke når man bare har to årsaker? Kunne jo like gjerne vært en if/else eller en ternary. Passer jo bra med de andre i fila men jeg er likevel usikker